### ISSUE TITLE
Update doesnt work anymore

### ISSUE THREAD

**Issue by domi-cloud** [jakeswenson/BitBetter#207]
Hello,
can someone help me here please?
I'm getting this error-message when I am executing the update script:

ERROR: Version mismatch: file /opt/bitwarden/bwdata/docker/docker-compose.yml specifies version 1 but extension file /opt/bitwarden/bwdata/docker/docker-compose.override.yml uses version 3.0

**Comment by codefreak558** [comment]
For that you can either just ignore is or in the docker-compose.override.yml remove the version 3.0 at the top.

Also if your bitwarden doesn't start up I found that downgrading bitwarden by setting the version to 2024.8.0 in both BitBetter's build.sh file and your bitwarden.sh file. Then pull both and start bitwarden and it should work.

It seems there might be a bug in 2024.8.1 where SSO and Admin are not able to connect to the mssql container.

And then finnaly update the Dockerfile as follows
```
ARG BITWARDEN_TAG=latest
FROM ${BITWARDEN_TAG}

COPY bin/Release/net8.0/publish/* /bitBetter/
COPY ./.keys/cert.cert /newLicensing.cer

RUN set -e; set -x; \
    dotnet /bitBetter/bitBetter.dll && \
    mv /app/Core.dll /app/Core.orig.dll && \
    mv /app/modified.dll /app/Core.dll && \
    rm -rf /bitBetter && rm -rf /newLicensing.cer
```

**Comment by codefreak558** [comment]
After I had changed the versions to 2024.8.0 I ran the build for BitBetter and then started up bitwarden in my installation folder so that it would pull the older version.

For the bitwarden.sh file I only changed the server to 2024.8.0 and left the other two version as they were as they worked.

**Comment by domi-cloud** [comment]
So i will set COREVERSION to 2024.8.0 in the bitwarden.sh file, BW_VERSION to 2024.8.0 in the build.sh file, then run the build script, start bitwarden and finally update the docker file while bitwarden is running?

> After I had changed the versions to 2024.8.0 I ran the build for BitBetter and then started up bitwarden in my installation folder so that it would pull the older version.
> 
> For the bitwarden.sh file I only changed the server to 2024.8.0 and left the other two version as they were as they worked.

So i will set COREVERSION to 2024.8.0 in the bitwarden.sh file, BW_VERSION to 2024.8.0 in the build.sh file, then run the build script, start bitwarden and finally update the docker file while bitwarden is running?

**Comment by domi-cloud** [comment]
> After I had changed the versions to 2024.8.0 I ran the build for BitBetter and then started up bitwarden in my installation folder so that it would pull the older version.
> 
> For the bitwarden.sh file I only changed the server to 2024.8.0 and left the other two version as they were as they worked.

So i will set COREVERSION to 2024.8.0 in the bitwarden.sh file, BW_VERSION to 2024.8.0 in the build.sh file, then run the build script, start bitwarden and finally update the docker file while bitwarden is running?

**Comment by domi-cloud** [comment]
> That should work, I would make a backup of your bitwarden installation folder just as an incase but it should work as that is what I did and that had worked for me.

Starting bitwarden after setting the variables and executing the build script, produces the following output:

| |__ () |__ ____ _ _ __ _| | ___ _ __
| ' | | \ \ /\ / /  | '__/ _ |/ _ \ '
| |) | | | \ V V / (| | | | (| | / | | |
|_./||_| _/_/ _,|| _,_|_|| ||

Open source password management solutions
Copyright 2015-2024, 8bit Solutions LLC
https://bitwarden.com/, https://github.com/bitwarden

===================================================

WARNING: This script is running as the root user!
If you are running a standard deployment this script should be running as a dedicated Bitwarden User as per the documentation.
Do you still want to continue? (y/n): y
Continuing....
bitwarden.sh version 2024.8.0
Docker version 24.0.7, build 24.0.7-0ubuntu2~20.04.1
docker-compose version 1.25.0, build unknown

Pulling api (bitbetter/api:2024.1.0)...
ERROR: The image for the service you're trying to recreate has been removed. If you continue, volume data could be lost. Consider backing up your data before continuing.

Continue with the new image? [yN]
ERROR: pull access denied for bitbetter/api, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

**Comment by GieltjE** [comment]
Probably fixed with pull request #213

**Comment by domi-cloud** [comment]
> Probably fixed with pull request #213

Hi GieltjE,
so how is the procedure now?
Thank you

**Comment by codefreak558** [comment]
Did get a email from my self hosted instance that my license had expired however as I had already made a organization it didn't affect me however when I went to try and build the licenseGen application I got the following error:
![image](https://github.com/user-attachments/assets/00ab7ab1-e61b-4406-91d8-9b89d3de248b)

**Comment by codefreak558** [comment]
It looks like once you update your bitbetter image for me I had to go and regenerate the license key's for my organizations as well as users who were not in a organization.

Your best bet is to look pull request #215 as I did a small change to the dockerfile so that you can actually build the image.

**Comment by h44z** [comment]
should be fixed by #217
